diff -ur kdelibs-4.0.2/doc/common/CMakeLists.txt kdelibs-4.0.2-install-all-css/doc/common/CMakeLists.txt
--- kdelibs-4.0.2/doc/common/CMakeLists.txt	2008-01-05 01:00:05.000000000 +0100
+++ kdelibs-4.0.2-install-all-css/doc/common/CMakeLists.txt	2008-03-06 05:18:11.000000000 +0100
@@ -7,6 +7,8 @@
     xml.dcl
     doxygen.css
     tabs.css
+    kde.css flat.css print.css
+    Doxyfile.global
     header.html
     footer.html
     mainheader.html
